src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -15,14 +15,21 @@ async function get_object(req, res) {
     const noobaa_trigger_agent = agent_header && agent_header.includes('exec-env/NOOBAA_FUNCTION');
     const encryption = s3_utils.parse_encryption(req);
 
-    const object_md = await req.object_sdk.read_object_md({
+    let get_from_cache = false;
+    if (req.query.get_from_cache === 'true') {
+        get_from_cache = true;
+    }
+    let md_params = {
         bucket: req.params.bucket,
         key: req.params.key,
         version_id: req.query.versionId,
         md_conditions: http_utils.get_md_conditions(req),
         encryption
-    });
-
+    };
+    if (get_from_cache) {
+        md_params.get_from_cache = true;
+    }
+    const object_md = await req.object_sdk.read_object_md(md_params);
```

Comment:
ok

Reply:
@guymguym I have to take back my previous answer.  If we made the changes you proposed, then we would have to change other namespace files (e,g, namespace_nb.js) for removing get_from_cache from params passed to API read_object_md and read_object_stream, since get_from_cache is not in the API schema.
Even after replacing ```get_from_cache: req.query.get_from_cache === 'true'``` with ```get_from_cache: req.query.get_from_cache === 'true' ? true : undefined```, it will not work either without changes in other places, since rpc schema validation takes undefined field as present.

Reply:
OK so our schemas are strict and we inject `additionalProperties: false` by default to all of our schemas.
See strictify in schema_utils:
https://github.com/noobaa/noobaa-core/blob/9fe155f0191742fc0ee4d4eaad1da79fec47622a/src/util/schema_utils.js#L137

This causes ajv to fail if the key exists even if it is undefined.
Opened an issue on AJV - https://github.com/ajv-validator/ajv/issues/1240
Resolving this comment for now until we have a better way.